src/compile/mark/mixins.ts

```diff
@@ -283,17 +286,26 @@ export function centeredBandPosition(channel: 'x' | 'y', model: UnitModel, defau
   };
 }
 
-export function binnedPosition(fieldDef: FieldDef<string>, channel: 'x'|'y', scaleName: string, spacing: number, reverse: boolean) {
-  if (channel === 'x') {
+export function binPosition(fieldDef: FieldDef<string>, fieldDef2: ValueDef | FieldDef<string>, channel: 'x'|'y', scaleName: string, spacing: number, reverse: boolean) {
+  const binSpacing = {
+    x:  reverse ? spacing : 0,
+    x2: reverse ? 0 : spacing,
+    y:  reverse ? 0 : spacing,
+    y2: reverse ? spacing : 0
+  };
+  if (isBinning(fieldDef.bin) && fieldDef2 === undefined) {
     return {
-      x2: ref.bin(fieldDef, scaleName, 'start', reverse ? 0 : spacing),
-      x: ref.bin(fieldDef, scaleName, 'end', reverse ? spacing : 0)
+      [channel]: ref.bin(fieldDef, scaleName, 'end', binSpacing[channel]),
+      [`${channel}2`]: ref.bin(fieldDef, scaleName, 'start', binSpacing[`${channel}2`]),
```

Comment:
better declare `const channel2 = channel === 'x' ? 'x2' : 'y2';` is better esp if we add stricter return type in the future.

Reply:
Done